planning/behavior_path_planner/autoware_behavior_path_planner_common/test/test_static_drivable_area.cpp

```diff
@@ -0,0 +1,541 @@
+    const auto result = cutOverlappedLanes(path, lanes);
+    ASSERT_EQ(result.size(), lanes.size());
+    EXPECT_TRUE(equal(result.front(), lanes.front()));
+    ASSERT_EQ(path.points.size(), path_size);
+    for (auto i = 0UL; i < path_size; ++i) {
+      EXPECT_EQ(path.points[i].point.pose.position.x, i * 1.0);
+      EXPECT_EQ(path.points[i].point.pose.position.y, 0.0);
+    }
+  }
+  {  // add more drivable lanes without an overlap (no overlap -> path is not modified)
+    reset_path();
+    lanes.push_back(
+      make_drivable_lanes(make_lanelet({2.0, 1.0}, {4.0, 1.0}, {2.0, -1.0}, {4.0, -1.0})));
+    lanes.push_back(
+      make_drivable_lanes(make_lanelet({4.0, 1.0}, {6.0, 1.0}, {4.0, -1.0}, {6.0, -1.0})));
+    const auto result = cutOverlappedLanes(path, lanes);
+    ASSERT_EQ(result.size(), lanes.size());
+    for (auto i = 0UL; i < result.size(); ++i) {
+      EXPECT_TRUE(equal(result[i], lanes[i]));
+    }
+    ASSERT_EQ(path.points.size(), path_size);
+    for (auto i = 0UL; i < path_size; ++i) {
+      EXPECT_EQ(path.points[i].point.pose.position.x, i * 1.0);
+      EXPECT_EQ(path.points[i].point.pose.position.y, 0.0);
+    }
+  }
+  {  // add an overlapping lane
+    reset_path();
+    lanes.push_back(
+      make_drivable_lanes(make_lanelet({2.5, -1.0}, {2.5, 1.0}, {3.5, -1.0}, {3.5, 1.0})));
+    const auto result = cutOverlappedLanes(path, lanes);
+    // the last lane is cut
+    ASSERT_EQ(result.size() + 1, lanes.size());
+    for (auto i = 0UL; i < result.size(); ++i) {
+      EXPECT_TRUE(equal(result[i], lanes[i]));
+    }
+    // since the path points do not have ids, all points are cut
+    EXPECT_TRUE(path.points.empty());
+  }
+  {  // add the overlapping lane id to the path points
+    reset_path();
+    for (auto & p : path.points) {
+      p.lane_ids.push_back(lanes.back().left_lane.id());
+    }
+    cutOverlappedLanes(path, lanes);
+    // since the overlapped lane was removed, the path points were still cut
+    EXPECT_TRUE(path.points.empty());
+  }
+  {  // add the first lane id to some path points, only these points will be kept
+    reset_path();
+    constexpr auto filtered_start = 3UL;
+    constexpr auto filtered_size = 5UL;
+    for (auto i = filtered_start; i < filtered_start + filtered_size; ++i) {
+      path.points[i].lane_ids.push_back(lanes.front().left_lane.id());
+    }
+    cutOverlappedLanes(path, lanes);
+    ASSERT_EQ(path.points.size(), filtered_size);
+    for (auto i = 0UL; i < filtered_size; ++i) {
+      EXPECT_EQ(path.points[i].point.pose.position.x, (i + filtered_start) * 1.0);
+      EXPECT_EQ(path.points[i].point.pose.position.y, 0.0);
+    }
+  }
+}
+
+TEST(StaticDrivableArea, generateDrivableLanes)
+{
+  using autoware::behavior_path_planner::utils::generateDrivableLanes;
+  lanelet::ConstLanelets lanelets;
+  lanelet::Lanelet ll;
+  {
+    const auto lanes = generateDrivableLanes(lanelets);
```

Comment:
nits: maybe you can add some comments here as you did for other TEST

Reply:
Done: 80aa2cc8a4